backend/test/private-api/private-api.notes.e2e-spec.ts

```diff
@@ -0,0 +1,1067 @@
+        'test.png',
+        testImage,
+        testSetup.userIds[0],
+        testSetup.ownedNoteIds[0],
+      );
+    }
+    describe('return media upload', () => {
+      beforeEach(async () => {
+        await uploadMedia();
+      });
+      afterEach(async () => {
+        await fs.unlink(join(uploadPath, upload + '.png'));
+        await fs.rm(uploadPath, { recursive: true });
+      });
+      it('for the owner', async () => {
+        const responseAfter = await agentUser1
+          .get(`${PRIVATE_API_PREFIX}/notes/${noteAlias1}/media/`)
+          .expect('Content-Type', /json/)
+          .expect(200);
+        expect(responseAfter.body).toHaveLength(1);
+        expect(responseAfter.body[0].uuid).toEqual(upload);
+      });
+      it('for another user with read permissions', async () => {
+        const responseAfter = await agentUser2
+          .get(`${PRIVATE_API_PREFIX}/notes/${noteAlias1}/media/`)
+          .expect('Content-Type', /json/)
+          .expect(200);
+        expect(responseAfter.body).toHaveLength(1);
+        expect(responseAfter.body[0].uuid).toEqual(upload);
+      });
+      it('for a guest user with read permissions', async () => {
+        const responseAfter = await agentGuestUser
+          .get(`${PRIVATE_API_PREFIX}/notes/${noteAlias1}/media/`)
+          .expect('Content-Type', /json/)
+          .expect(200);
+        expect(responseAfter.body).toHaveLength(1);
+        expect(responseAfter.body[0].uuid).toEqual(upload);
+      });
+    });
+    describe('throws an error', () => {
+      it('if the user is not logged in', async () => {
+        await agentNotLoggedIn
+          .get(`${PRIVATE_API_PREFIX}/notes/${noteAlias1}/media/`)
+          .expect('Content-Type', /json/)
+          .expect(401);
+      });
+      it("if the user can't read the note", async () => {
+        await testSetup.permissionsService.removeGroupPermission(
+          testSetup.ownedNoteIds[0],
+          await testSetup.groupService.getGroupIdByName(SpecialGroup.EVERYONE),
+        );
+        await testSetup.permissionsService.removeGroupPermission(
+          testSetup.ownedNoteIds[0],
+          await testSetup.groupService.getGroupIdByName(SpecialGroup.LOGGED_IN),
+        );
+        await agentUser2
+          .get(`${PRIVATE_API_PREFIX}/notes/${noteAlias1}/media/`)
+          .expect('Content-Type', /json/)
+          .expect(403);
+      });
+      it('when the note does not exist', async () => {
+        await agentUser1
+          .get(`${PRIVATE_API_PREFIX}/notes/i_dont_exist/media/`)
+          .expect('Content-Type', /json/)
+          .expect(404);
+      });
+      it('fails for a forbidden note', async () => {
+        await agentUser1
+          .get(`${PRIVATE_API_PREFIX}/notes/${forbiddenNoteId}/media`)
+          .expect('Content-Type', /json/)
+          .expect(403);
+      });
+    });
+  });
+
+  describe(`POST ${PRIVATE_API_PREFIX}/notes`, () => {
+    describe('can create a new note', () => {
```

Comment:
For matching sentence rule in test naming it should be:

```suggestion
    describe('a new note can be created by', () => {
```